apps/vc-api/src/vc-api/credentials/credentials.service.ts

```diff
@@ -54,101 +65,73 @@ interface ISpruceVerifyOptions {
+      format: ClaimFormat.LdpVc,
+      proofType: 'Ed25519Signature2018',
+      verificationMethod: verificationMethod.id
+    };
+    const w3cVerifiableCredential =
+      await this.credoService.agent.w3cCredentials.signCredential<ClaimFormat.LdpVc>(credentialOption);
+    return w3cVerifiableCredential.toJson() as VerifiableCredentialDto;
   }
 
   async verifyCredential(
     vc: VerifiableCredentialDto,
     options: VerifyOptionsDto
   ): Promise<VerificationResultDto> {
-    const verifyOptions: ISpruceVerifyOptions = options;
-
-    return didKitExecutor<VerificationResultDto>(
-      () => verifyCredential(JSON.stringify(vc), JSON.stringify(verifyOptions)),
-      'verifyCredential'
-    );
+    const w3cVerifyCredentialOptions: W3cVerifyCredentialOptions<ClaimFormat.LdpVc> = {
+      credential: W3cJsonLdVerifiableCredential.fromJson(vc) //JsonTransformer.fromJSON(vc, W3cJsonLdVerifiableCredential),
+    };
+    const obj = { ...w3cVerifyCredentialOptions };
+    const verifyCredential: W3cVerifyCredentialResult =
+      await this.credoService.agent.w3cCredentials.verifyCredential(obj as any);
+    return verifyCredential;
   }
 
   /**
    * Assembles presentation from provided credentials according to definition
    */
-  presentationFrom(
-    presentationDefinition: IPresentationDefinition,
-    credentials: VerifiableCredentialDto[]
-  ): PresentationDto {
-    const pex = new PEX();
-    // presentation should be created from selected credentials https://github.com/Sphereon-Opensource/pex/issues/91#issuecomment-1115940908
-    const { verifiableCredential, areRequiredCredentialsPresent } = pex.selectFrom(
-      presentationDefinition,
-      credentials as IVerifiableCredential[]
-    );
-    if (areRequiredCredentialsPresent !== Status.INFO) {
-      throw new InternalServerErrorException('Credentials do not satisfy defintion');
-    }
-    const presentation = pex.presentationFrom(presentationDefinition, verifiableCredential);
-
-    // embed submission context to workaround failing to load context 'https://identity.foundation/presentation-exchange/submission/v1'
-    const submissionContext = {
-      PresentationSubmission: {
-        '@id': 'https://identity.foundation/presentation-exchange/#presentation-submission',
-        '@context': {
-          '@version': '1.1',
-          presentation_submission: {
-            '@id': 'https://identity.foundation/presentation-exchange/#presentation-submission',
-            '@type': '@json'
-          }
-        }
-      }
-    };
-    const submissionContextUri = 'https://identity.foundation/presentation-exchange/submission/v1';
-
-    presentation['@context'] = Array.isArray(presentation['@context'])
-      ? presentation['@context']
-      : [presentation['@context']];
-
-    presentation['@context'] = presentation['@context'].filter((c) => c !== submissionContextUri);
-    presentation['@context'].push(submissionContext as any);
-    return presentation as PresentationDto;
```

Comment:
@mesaresachin My first impression is that this removal of usage of the `pex` library is the largest functional change in the PR. I suppose it because credo is struggling to add the proof to the presentation created by `pex`?

Reply:
Reverted to the original implementation and removed `submissionContext` from it as Credo does not recognize it.